pkg/kubelet/cm/dra/plugin/client.go

```diff
@@ -131,6 +132,9 @@ func (r *draPluginClient) NodePrepareResource(
 		ResourceHandle: resourceHandle,
 	}
 
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
 	return nodeClient.NodePrepareResource(ctx, req)
 }
```

Comment:
Isn't this just going to cancel the context as soon as this function returns? Or is that the intent?

I guess this makes sense so long as `nodeClient.NodePrepareResource()` is blocking. Meaning either the (blocking) call to `nodeClient.NodePrepareResource()` succeeds and we now want to cancel the context, or it times out, at which point the context is already canceled and calling `cancel()` directly is then a noop.

@bart0sh can you confirm that `nodeClient.NodePrepareResource()` is blocking and operates as described above?

Reply:
Yes, `nodeClient.NodePrepareResource()` a blocking call.